pkg/io/test/marshalertest.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package test contains the generic serializer tests.
+package test // import "perun.network/go-perun/pkg/io/test"
+
+import (
+	"encoding"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type binary interface {
+	encoding.BinaryMarshaler
+	encoding.BinaryUnmarshaler
+}
+
+// GenericMarshalerTest runs multiple tests to check whether encoding
+// and decoding of serializer values works.
+func GenericMarshalerTest(t *testing.T, serializers ...binary) {
```

Comment:
Don't we already have something like that? @ggwpez

Reply:
Is this MR rebased onto latest `dev`? Package `pkg` got moved to https://github.com/perun-network/poly-go.  
Besides that; Yes. Its called `GenericSerializerTest`, but probably does not work for your `binary` interface?

Reply:
Reg `pkg/io`, 
no this PR is not rebased onto latest dev. Myself and @matthiasgeihs had a discussion regarding `pkg/io` and figured out that it is better to move it back into go-perun. Because it is very specific to perun, unlike other packages in `pkg`.
So, I will have wait until `pkg/io` is moved back into `go-perun` before rebasing.

Reg `GenericSerializerTest`, as you pointed out, it doesn't work for `binary` interface. Since, we will be changing some more types to implement binary interface (instead of Encoder), I added this test.